dask/dataframe/tests/test_io.py

```diff
@@ -547,11 +547,10 @@ def test_read_csv_of_modified_file_has_different_name():
     with filetext(text) as fn:
         mtime = os.path.getmtime(fn)
         a = read_csv(fn)
+        sleep(1)
         with open(fn, 'a') as f:
             f.write('\nGeorge,700')
             os.fsync(f)
-        while mtime == os.path.getmtime(fn):
-            sleep(0.01)
```

Comment:
One solution would be to do the one second sleep if the operating system is OSX and then mark the test as slow (also if on OSX).